lesson7-questions.txt

```diff
@@ -87,19 +117,39 @@ You will see the SQL that is executed logged to the screen.  What is that SQL?
 There is a security exposure, a risk that a hacker could do damage.  What is that
 risk? What two things are done in your code to prevent the risk?
 
+The risk is that unauthorized user can delete the post.
+
+One thing that prevennt the risk is:
+  before_action :check_logon, except: %w[show] 
+Which checks if the user is logged in before any action except showing the forum content
+
+One more thing is:
+  before_action :check_access, only: %w[edit update destroy] 
+Which checks if the logged inn user have access to edit, update, or delete post.
 
 
 9. The routes for subscriptions are nested under the forum routes.  How does this
 change the routes?  What is the additional hash key and value that is passed in the
 params hash to the subscriptions controller as a result of this nesting? Which routes
 are unaffected by the nesting?
 
+When the routs are nested, the information about parennt rout is included in the URL. Example:
+  POST /forums/:forum_id/subscriptions => subscriptions#create
 
+:forum_id hashs key with the value forum.id is passed in the
+params hash to the subscriptions controller as a result of this nesting.
+
+Using shallow: true, except: [:index] allows to leave #show, #edit, #update, and #destroy routs unaffected by the nesting.
```

Comment:
Shallow Nesting: When you use shallow: true, only the new, create, and index actions will remain nested under forum. This means they still require the forum_id in the URL. For example, POST /forums/:forum_id/subscriptions would be used to create a new subscription.

On the other hand, actions like show, edit, update, and destroy will use non-nested URLs, meaning they will not include :forum_id in the URL. For instance, GET /subscriptions/:id will be used to show a subscription, and this route doesn't require the forum_id.